pkg/retester/retester_test.go

```diff
@@ -443,3 +448,128 @@ func TestEnabledPRs(t *testing.T) {
 		})
 	}
 }
+
+func TestLoadFromDisk(t *testing.T) {
+	logger := logrus.NewEntry(logrus.StandardLogger())
+	testCases := []struct {
+		name           string
+		file           string
+		cacheRecordAge string
+		bytes          []byte
+		perm           fs.FileMode
+		expectedError  error
+	}{
+		{
+			name:           "basic case",
+			file:           "backoff.cache",
+			cacheRecordAge: "24h",
+			bytes: []byte(`pr:
+  last_considered_time: "2200-01-01T00:00:00Z"`),
+			perm:          0644,
+			expectedError: nil,
+		},
+		{
+			name:           "empty file",
+			file:           "",
+			cacheRecordAge: "24h",
+			perm:           0644,
+			expectedError:  nil,
+		},
+		{
+			name:           "file no exist",
+			file:           "no-exist.cache",
+			cacheRecordAge: "24h",
+			perm:           0644,
+			expectedError:  nil,
+		},
+		{
+			name:           "file no read perm",
+			file:           "backoff.cache",
+			cacheRecordAge: "24h",
+			perm:           0000,
+			expectedError:  errors.New("failed to read file backoff.cache: open backoff.cache: permission denied"),
+		},
+		{
+			name:           "wrong formatting",
+			file:           "backoff.cache",
+			cacheRecordAge: "24h",
+			bytes: []byte(`wrong:
+			formating"`),
+			perm:          0644,
+			expectedError: errors.New("failed to unmarshal: error converting YAML to JSON: yaml: line 2: found character that cannot start any token"),
+		},
+		{
+			name:           "old case",
+			file:           "backoff.cache",
+			cacheRecordAge: "24h",
+			bytes: []byte(`pr:
+  last_considered_time: "1970-01-01T00:00:00Z"`),
+			perm:          0644,
+			expectedError: nil,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			defer func() {
+				tmp := "backoff.cache"
```

Comment:
The tmp file should be in tmp folder.
https://yourbasic.org/golang/temporary-file-directory/